src/Sema.zig

```diff
@@ -36513,10 +36478,12 @@ fn semaUnionFields(pt: Zcu.PerThread, arena: Allocator, union_ty: InternPool.Ind
     var comptime_err_ret_trace = std.ArrayList(LazySrcLoc).init(gpa);
     defer comptime_err_ret_trace.deinit();
 
-    var sema: Sema = .{
+    const cau_index = union_type.cau;
+
+    var inner_sema: Sema = .{
```

Comment:
Why do we need this? I think you may have written this before you made `Type.resolveUnionInner` set `code`.